frontend/apps/reader/modules/readerfont.lua

```diff
@@ -340,6 +340,10 @@ function ReaderFont:addToMainMenu(menu_items)
             return T(_("Font: %1"), BD.wrap(self.font_face))
         end,
         sub_item_table_func = function()
+            -- move currently set font to the top of the list if sorted by recently selected
+            if G_reader_settings:isTrue("font_menu_sort_by_recently_selected") then
+                self:addToRecentlySelectedList(self.font_face)
+            end
```

Comment:
I don't think this is the right place to do that: it's odd that some setting change (or reordering) would happen when one opens a menu :)
But it seems that is what you wanted:
> Ideally I'd like the current font move to the top position only when the Font menu is opened (and then the font list immediately refreshed); not when the document is opened. 

I'm not sure I understand why you would want that - and not when the document is opened.

Reply:
Glad you're back :)

>I'm not sure I understand why you would want that - and not when the document is opened.

It is to avoid reordering the font list when the user doesn't intend to change the font (or changes it only outside the menu; using a profile or quick menu). 

Otherwise, if the user frequently switches between different books that have different fonts, the font list gets reordered each time, even though he never actively changed the font. 

The purpose for this reordering is only so that when the user opens the menu, he doesn't find himself somewhere on page 10 of the font list and first has to go back to page 1 to select one of his most recently selected fonts.

Reply:
So, if one opens 3 books (having each a different font), and he opens the menu, on the next book opening, he will get these 3 fonts first as recently selected, _even though he never actively changed the font_ ? :/
I think you should not touch the recently_selected list.

You probably should try to reorder face_list to move the current font at the first position in between here:
https://github.com/koreader/koreader/blob/75d3e4c2dcc184281fff490431e98123b6ba0c26/frontend/apps/reader/modules/readerfont.lua#L72-L73
so this affects only the menu table and the UI.

Reply:
Yes, this makes sense and I was actually trying this before, but am struggling with the same problem again: how do I get the currently set font at this stage? `self.font_face` is not defined yet and `G_reader_settings:readSetting("cre_font")` apparently returns the default font? What do I call here to get the font?

Reply:
You could try to move the building of the menu table to later when the settings are read:
```diff
--- a/frontend/apps/reader/modules/readerfont.lua
+++ b/frontend/apps/reader/modules/readerfont.lua
@@ -32,5 +32,4 @@ local newly_added_fonts = nil -- not yet filled
 function ReaderFont:init()
     self:registerKeyEvents()
-    self:setupFaceMenuTable()
     self.ui.menu:registerToMainMenu(self)
     -- NOP our own gesture handling
@@ -185,4 +184,6 @@ function ReaderFont:onReadSettings(config)
     self:updateFontFamilyFonts()

+    self:setupFaceMenuTable()
+
     -- Dirty hack: we have to add following call in order to set
     -- m_is_rendered(member of LVDocView) to true. Otherwise position inside
```
I don't think anything from it is needed before that - but it is to be tested and confirmed by you :)
(The alternative would be to check `if self.font_face then do your stuff end`, so not have it done on the first call - and later re-call it if font_menu_sort_by_recently_selected=true.)

Reply:
Thank you! It works like a charm! Now I just have to figure out how to get rid of the uninvited guests in my PR :/. The problem was that those changes were selected by default in my Github Desktop. I should have unchecked them before committing anything, but I didn't see that.